### PR TITLE
Add D&D Beyond Importer spell automations/macros

### PR DESCRIPTION
**Pull request by TheGiddyLimit**
Here we fill in gaps left by CPR/MidiSRD, by grabbing available automations from the D&D Beyond Importer module. We do this, instead of depending on the module directly, for two reasons:

- DDBI, in code terms, is structured in such a way as to make pulling these automations out at runtime difficult. Viable options are limited to exceedingly flaky/dangerous big-`eval`s, or, having the user import everything from D&D Beyond in advance ( :^) ).
- DDBI represents a, to put it mildly, alternate workflow for importing content. Forcing the user into running both is bloaty at best.

As DDBI is regularly updated, keeping especially the macros in-sync is a task which will require future maintenance. Hopefully as other integrations advance, some of these spells will be automated elsewhere, and we can prune down the list of copies we're maintaining. In the meantime, additions have been made to the converter page to facilitate easy export of `system` data, and of item macros*.

In particular, some DDBI macros make use of a DDBI module API; a 1:1 port of this has therefore been made here (with renames to avoid any current and future collisions). As the references to this API are minimal and easy to find-replace, this solution seems "good enough" for the moment.

Regardless of potential future maintenance, various spells have seen basic automation improvements which should make for an all-round better experience moving forward.

* A future migration should likely be done to transition towards the new DAE "DIME" system, but this is work for another PR.

### CODE REVIEW

**Comment by Spappz** [comment]
I agree re: avoiding a dependency on DDBI. Even ignoring bloat, encouraging people to use DDBI probably opens a whole can of worms in terms of Plutonium compatibility (I believe DDBI includes its own 5e character sheet? Or at least modifies it in some way, and I imagine DDBI will import things lacking Plutonium metadata so things like the level-up button will start to falter).

> As DDBI is regularly updated, keeping especially the macros in-sync is a task which will require future maintenance. Hopefully as other integrations advance, some of these spells will be automated elsewhere, and we can prune down the list of copies we're maintaining.

One slightly sneaky possibility is to introduce a basic-level version into CPR or MidiSRD rather than here, and then simply palm it off to those maintainers to maintain 😏

I'm a little surprised many of these aren't already included by either of the others.

---

I'm a little pressed for time right now, but after consulting Midi's README (Posney's been adding things 😩) at one out of the effect I checked looks good.  I'll get back to this sometime Soon™ (or if you're confident it Just Works™ then feel free to merge anyway)

**Comment by revilowaldow** [comment]
If this is working Sufficiently ™️ then I guess no need to change it, but this does feel pretty rubbish.
There's the heuristic option, with lots of complicated remainder finding etc., or we could just run the incoming scripts through our pre-existing js linter prior to use? Then we'd know their indent styles. Further to that point, if these are direct copies, maybe they should be minified. Any fault within the files would probably just get reported upstream? 

image

**Comment by TheGiddyLimit** [comment]
> if you're confident it Just Works:tm:

Just about; I went through and tested each one (as some tweaking was required to get Plutonium's output to line up with the expected targeting/hooks), so they really _should_ all be working as expected (except the Spike Growth one, which has a `FIXME`--keeping it in for now, as more obvious having it there but broken rather than silently missing)

---

re: world's worst heuristic, this was actually _because_ the linter was doing a crummy job of reformatting, and doing a little janky pre-cleaning made it smoother :^) It can be improved whenever, but for the moment, will ensure re-runs vs. the same scripts (i.e. whenever I next dump DDB's stuff to update what's here) will have minimal diffs

tl;dr: at best it helps, and at worst it doesn't do much, and I'd prefer not to have to hack the same bodge in every time I want to run the thing

if you insist I can spend the time to make it less arse, but YAGNI etc.

---

re: minification, not a bad idea in theory, but I _did_ have to debug around in these when I was testing them, and that took long enough as-is without having to work back from un-minified rubbish. Do note that the header comment saying "property of DDBI" does _not_ get imported to the final item macro, so there'll be no way for Johnny Enduser to know where they came from unless they really go looking (and I hope that anyone with enough technical knack to do so would create a bug report on _our_ repo/Discord, rather than angrily storming off)

**Comment by revilowaldow** [comment]
All sounds good to me 👍
Keep up the good work mate

**Comment by Spappz** [comment]
> re: world's worst heuristic

Vauxs and I have been enjoying the simplicity that Prettier offers, if you do want consistent, non-minified code 😏

**Comment by TheGiddyLimit** [comment]
fair point, I'll try breaking out the big guns
